src/main/java/rx/internal/schedulers/CachedThreadScheduler.java

```diff
@@ -190,7 +190,14 @@ public Worker createWorker() {
         public void unsubscribe() {
             if (once.compareAndSet(false, true)) {
                 // unsubscribe should be idempotent, so only do this once
-                pool.release(threadWorker);
+
+                // Release the worker _after_ the previous action (if any) has completed
+                threadWorker.schedule(new Action0() {
```

Comment:
It saves an allocation; I believe io() workers are used and unsubscribed frequently enough to save on this allocation.